reducers/index.js

```diff
@@ -5,26 +5,28 @@ module.exports = function navigate(state, {type, payload}) {
     return {};
   }
 
-  const stack = state.stack;
-  const index = state.index;
+  const stack = state.__navRedux.stack;
+  const index = state.__navRedux.index;
 
   switch (type) {
     case PUSH:
       return Object.assign({}, state, {
-        stack: stack.unshift(payload),
-        index: 0,
+        __navRedux: {
+          stack: stack.push(payload),
+          index: 0,
+        },
       });
 
     case POP:
-      const incremented = index + 1;
-
-      if (incremented === stack.count()) {
+      if (stack.count() === 1) {
         return state;
```

Comment:
Maybe it is not a good idea to throw an error here.
Because if the user has only one route in the navigator and call `pop()`
The expected behaviour is: do nothing.

To be even better we can have in our state a flag `canBack` `canForward` so the user is able to render his buttons properly